pennylane/transforms/core/transform_program.py

```diff
@@ -354,24 +352,32 @@ def set_classical_component(self, qnode, args, kwargs):
             self._set_all_classical_jacobians(qnode, args, kwargs, argnums)
             self._set_all_argnums(qnode, args, kwargs, argnums)
 
-    def prune_dynamic_transform(self):
-        """Ensure a single ``dynamic_one_shot`` transform is applied."""
-        trans_type = np.zeros(len(self._transform_program), dtype=np.int32)
-        for i, t in enumerate(self._transform_program):
-            if "dynamic_one_shot" in str(t):
-                trans_type[i] = 1
-            if "mid_circuit_measurements" in str(t):
-                trans_type[i] = 2
-        if sum(trans_type) < 2:
-            return
-        keep = 2 if 2 in trans_type else 1
+    def _prune_dynamic_transform(self, type_to_keep=1):
```

Comment:
Given this is a public method on a key architectual component, I don't think we should rename/ remove it without a deprecation cycle.  @vincentmr any thoughts?

Reply:
I figured that it wasn't public on purpose, since it is only used internally in `qml.execute` to make sure that we don't end up having duplicate `dynamic_one_shot` transforms. The only time that this would happen is when the user manually applies `dynamic_one_shot` on a circuit, and device preprocessing adds another `dynamic_one_shot`, which is only internal.

Reply:
So there's public as in "user facing" , and public as in "accessible outside the class".

Reply:
True, I can make this public again.